google-common/src/main/scala/org/apache/pekko/stream/connectors/google/util/AnnotateLast.scala

```diff
@@ -45,14 +45,33 @@ private[google] final case class Last[+T](value: T) extends MaybeLast[T] {
 @InternalApi
 private[google] object AnnotateLast {
 
+  private final def accumulateState[T](maybePreviousElement: Option[T], elem: T) =
+    maybePreviousElement match {
+      case Some(previousElem) => (Some(elem), Some(NotLast(previousElem)))
+      case None               => (Some(elem), None)
+    }
+
   def apply[T]: Flow[T, MaybeLast[T], NotUsed] =
     Flow[T]
-      .statefulMap(() => Option.empty[T])((maybePreviousElement, elem) => {
-          maybePreviousElement match {
-            case Some(previousElem) => (Some(elem), Some(NotLast(previousElem)))
-            case None               => (Some(elem), None)
-          }
-        }, _.map(elem => Some(Last(elem))))
+      .statefulMap(() => Option.empty[T])(
+        accumulateState,
+        _.map(elem => Some(Last(elem)))
+      ).collect {
+        case Some(elem) => elem
+      }
+
+  def apply[T](zero: T): Flow[T, MaybeLast[T], NotUsed] =
+    Flow[T]
+      .statefulMap(() => Option.empty[T])(
+        accumulateState,
+        {
```

Comment:
Maybe the old style is simpler than the anymouse one?

Reply:
I prefer this one as there is less boilerplate

Reply:
that's ture